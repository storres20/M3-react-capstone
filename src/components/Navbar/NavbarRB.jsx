import { NavLink } from 'react-router-dom';
import {
  Navbar, Container, Nav,
} from 'react-bootstrap';
import styles from './NavbarRB.module.css';

import Designby from '../Designby/Designby';

export default function NavbarRB() {
  return (
    <Navbar bg="dark" expand="md" sticky="top" variant="dark">
      <Container>
        <Navbar.Brand className={styles.NavTitle} data-testid="nav-brand">
          <NavLink to="/">&#60;</NavLink>
          <NavLink to="/">Movies</NavLink>
        </Navbar.Brand>
        <Navbar.Toggle aria-controls="basic-navbar-nav" />
        <Navbar.Collapse id="basic-navbar-nav">
          <Nav className={`me-auto ${styles.NavBar}`} data-testid="nav-links">
            <NavLink to="/" className={styles.NavSubTitle}>Home</NavLink>
            <NavLink to="/about" className={styles.NavSubTitle}>About</NavLink>
          </Nav>
          <Designby />
        </Navbar.Collapse>
      </Container>
    </Navbar>
  );
}
