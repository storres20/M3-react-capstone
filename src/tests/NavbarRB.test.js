import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import NavbarRB from '../components/Navbar/NavbarRB';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <NavbarRB />
      </BrowserRouter>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
