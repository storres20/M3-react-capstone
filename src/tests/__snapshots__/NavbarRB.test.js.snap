// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<nav
  className="navbar navbar-expand-md navbar-dark bg-dark sticky-top"
>
  <div
    className="container"
  >
    <span
      className="NavTitle navbar-brand"
      data-testid="nav-brand"
    >
      <a
        aria-current="page"
        className="active"
        href="/"
        onClick={[Function]}
      >
        &lt;
      </a>
      <a
        aria-current="page"
        className="active"
        href="/"
        onClick={[Function]}
      >
        Movies
      </a>
    </span>
    <button
      aria-controls="basic-navbar-nav"
      aria-label="Toggle navigation"
      className="navbar-toggler collapsed"
      onClick={[Function]}
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      aria-expanded={null}
      className="navbar-collapse collapse"
      id="basic-navbar-nav"
    >
      <div
        className="me-auto NavBar navbar-nav"
        data-testid="nav-links"
        onKeyDown={[Function]}
      >
        <a
          aria-current="page"
          className="NavSubTitle active"
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
        <a
          className="NavSubTitle"
          href="/about"
          onClick={[Function]}
        >
          About
        </a>
      </div>
      <div
        className="designContainer"
        data-testid="designby"
      >
        <p>
          Designed by
           
          <a
            href="https://myport20.netlify.app/"
            rel="noreferrer"
            target="_blank"
          >
            <b>
              ITALO
            </b>
          </a>
        </p>
      </div>
    </div>
  </div>
</nav>
`;
